api/jsonapi.go

```diff
@@ -1139,30 +1139,87 @@ func (i *jsonAPIHandler) GETFollowing(w http.ResponseWriter, r *http.Request) {
 }
 
 func (i *jsonAPIHandler) GETInventory(w http.ResponseWriter, r *http.Request) {
-	type inv struct {
-		Slug     string `json:"slug"`
-		Variant  int    `json:"variant"`
-		Quantity int    `json:"quantity"`
+	// Get optional peerID and slug parameters
+	var (
+		peerIDString string
+		slug         string
+	)
+
+	parts := strings.Split(r.URL.Path, "/")
+	if len(parts) > 3 {
+		peerIDString = parts[3]
 	}
-	var invList []inv
-	inventory, err := i.node.Datastore.Inventory().GetAll()
+	if len(parts) > 4 {
+		slug = parts[4]
+	}
+
+	// If we want our own inventory get it from the local database and return
+	getPersonalInventory := (peerIDString == "" || peerIDString == i.node.IpfsNode.Identity.Pretty())
+	if getPersonalInventory {
+		type inv struct {
+			Slug     string `json:"slug"`
+			Variant  int    `json:"variant"`
+			Quantity int    `json:"quantity"`
+		}
+		var invList []inv
+		inventory, err := i.node.Datastore.Inventory().GetAll()
+		if err != nil {
+			fmt.Fprint(w, `[]`)
```

Comment:
@placer14 maybe, but this is the existing code for the /ob/inventory endpoint and there was no expected changes to its default behavior so I did not change it. If @rmisio thinks it's easy to handle we can but I don't want to change existing behavior with stakeholder buy-in.

Reply:
But this is probably not desired behavior. In an online transaction, wouldn't the buyer be told with this response "there are no inventory constraints, purchase as many as you like" but when the order is received with an inventory that's larger than possible, it would fail because the prior data was invalid?

Reply:
I agree that it's probably not desirable, but I'm not going to change behavior unrelated to what I'm working on unless the people using that behavior (@rmisio and @jjeffryes) are okay and aware of it.